src/components/settings/PermanentFilterSettings.tsx

```diff
@@ -0,0 +1,52 @@
+import gql from 'graphql-tag';
+import React, { useContext } from 'react';
+import { useQuery } from 'react-apollo';
+import { View } from 'react-native';
+
+import { graphqlFetchPolicy } from '../../helpers';
+import { usePermanentFilter, useRefreshTime } from '../../hooks';
+import { NetworkContext } from '../../NetworkProvider';
+import { FilterAction } from '../../types';
+import { SettingsToggle } from '../SettingsToggle';
+
+export const PermanentFilterSettings = () => {
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+
+  const refreshTime = useRefreshTime('settings-dataProviders');
+
+  const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp, refreshTime });
+
+  const { state, dispatch } = usePermanentFilter();
+
+  const { data } = useQuery(
+    gql`
+      query dataProviders {
+        newsItemsDataProviders {
+          id
+          name
+        }
+      }
+    `,
+    { fetchPolicy }
+  );
+
+  return (
+    <View>
+      {data?.newsItemsDataProviders?.map((item: { id: string; name: string }) => {
+        const options = {
+          title: item.name,
+          bottomDivider: true,
+          value: !state.includes(item.id),
+          onActivate: () => {
+            dispatch({ type: FilterAction.RemoveDataProvider, payload: item.id });
+          },
+          onDeactivate: () => {
+            dispatch({ type: FilterAction.AddDataProvider, payload: item.id });
+          }
+        };
```

Comment:
some scenario we can improve: when on the news index screen, in an app with drawer, i can directly navigate to the settings. if i change something and go back to the news items index, nothing changes. i need to go back to the homescreen and enter the news index again to have the refreshed stuff based on newly selected data providers. can we update the index screen as well on changes in settings?

Reply:
i adjusted the dependencies in the callbacks/effects to react to changes in the provider state